plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeFunctionProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import io.trino.filesystem.TrinoFileSystemFactory;
+import io.trino.plugin.deltalake.functions.tablechanges.TableChangesProcessorProvider;
+import io.trino.plugin.hive.parquet.ParquetReaderConfig;
+import io.trino.spi.function.FunctionProvider;
+import io.trino.spi.function.SchemaFunctionName;
+import io.trino.spi.ptf.TableFunctionProcessorProvider;
+
+import javax.inject.Inject;
+
+import static io.trino.plugin.deltalake.functions.tablechanges.TableChangesFunction.TABLE_CHANGES_NAME;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeFunctionProvider
+        implements FunctionProvider
+{
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final DeltaLakeConfig deltaLakeConfig;
+    private final ParquetReaderConfig parquetReaderConfig;
+
+    @Inject
+    public DeltaLakeFunctionProvider(TrinoFileSystemFactory fileSystemFactory, DeltaLakeConfig deltaLakeConfig, ParquetReaderConfig parquetReaderConfig)
+    {
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.deltaLakeConfig = requireNonNull(deltaLakeConfig, "deltaLakeConfig is null");
+        this.parquetReaderConfig = requireNonNull(parquetReaderConfig, "parquetReaderConfig is null");
+    }
+
+    @Override
+    public TableFunctionProcessorProvider getTableFunctionProcessorProvider(SchemaFunctionName name)
+    {
+        if (name.equals(TABLE_CHANGES_NAME)) {
+            return new TableChangesProcessorProvider(fileSystemFactory, deltaLakeConfig, parquetReaderConfig);
```

Comment:
Why do we have to create a new instance of `TableChangesProcessorProvider` at every method call?
I'd argue that `DeltaLakeFunctionProvider` needs to be rather injected.

Reply:
`DeltaLakeFunctionProvider` is injected. I guess you meant `TableChangesProcessorProvider` but then how do you imagine it would work ?
`TableChangesProcessorProvider` would be injected into `DeltaLakeFunctionProvider` and it would simply return it? It would also be strange imho.

Reply:
`DeltaLakeFunctionProvider` is expected to provide a `TableFunctionProcessorProvider` based on the a name of a function.
At the moment, we have only one table function and it seems that this class is a simple getter, but in case we'll be having multiple table functions, this aspect will change.

Reply:
ok, I don't like how it is going to look like but I will do it this way